test/integration/annotation/annotated_resource_setup.go

```diff
@@ -0,0 +1,318 @@
+package annotation
+
+import (
+	"context"
+	"fmt"
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/client"
+	"github.com/skupperproject/skupper/pkg/kube"
+	"github.com/skupperproject/skupper/pkg/utils"
+	"github.com/skupperproject/skupper/test/utils/k8s"
+	"gotest.tools/assert"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	nsCluster1 = "annotation-1"
+	nsCluster2 = "annotation-2"
+	timeout    = 120 * time.Second
+	interval   = 20 * time.Second
+)
+
+var (
+	clusters           int = 1
+	ctx                    = context.Background()
+	kubeConfigCluster1 string
+	kubeConfigCluster2 string
+	cluster1           *client.VanClient
+	cluster2           *client.VanClient
+	// teardownClusters prevents deleting an existing namespace
+	teardownClusters []*client.VanClient
+)
+
+// test allows defining the matrix to run the test table
+type test struct {
+	name                  string
+	doc                   string
+	modification          func(*testing.T)
+	expectedSites         int
+	expectedServicesProto map[string]string
+}
+
+// Setup simply creates the namespaces using a single
+// cluster by default, or using two clusters if at
+// least two kubeconfig files have been provided. It
+// also initializes the cluster1 and cluster2 variables.
+func Setup(t *testing.T) {
+	var err error
+
+	// Need at least one public (or a default) cluster
+	pubClusters := k8s.KubeConfigFilesCount(t, false, true)
+	if pubClusters == 0 {
+		t.Skip("no public clusters available")
+	}
+	if k8s.KubeConfigFilesCount(t, true, true) > 1 {
+		clusters = 2
+	}
+	t.Logf("Setup - using %d clusters", clusters)
```

Comment:
> * It seems to me you have created a new integration test from scratch, that does not integrate at all with the other existing integration tests.
> * It is using new cli options (to support multiple clusters), that are unknown to the other existing integration tests.

The reason why I have chosen not to use the ClusterTestRunnerBase is that is is currently statically using just one KUBECONFIG file, and I wanted to use it with more than one to be able to run this test across different cluster versions. 

Anyway I have already opened an issue to enhance the common (or base) test types used by the integration
tests in order to standardize the regular workflow #209 .


> * What I would do to avoid diverging and having to maintain two different testing trees is:
> * Strip this PR, removing al the logic related to supporting multiple clusters, and submit it as a separate pr that implements that for all existing integration tests.

Do you guys think it is better to do this now, instead of merging this one, which is supposed to work along with the
others (as long as we do not set the new flags) and working on #209 after that?


> * After the first item it would be not so difficult, to just use the existing ClusterTestRunner and ClusterContexts as the other tests are doing.
> * Finally and as separate PR, that closes the multicluster support issue, you can submit a PR that adds the multi-cluster support, modifying, removing, refactoring or anything you like to the "cluster" package, and that way that change will automatically applied to all integration tests.
> * What is in the cluster package right now if far from being a very beautiful piece of code I know ( I was learning Go while doing that) but also it is a tiny piece of code. I propose you to put your hands on it and modify, etc. As I have mentioned, instead of diverging, and creating a new integration test from scratch.

Anyway, I would be glad to work on #209 as my next assignment for Skupper.

Reply:
> t.Log prints all messages once the test have finished, is that what you want?

Actually it shows up immediately, as long as you run the test using "-test.v" and
unlike using fmt.Println which prints either way. See:

> func (c *B) Logf(format string, args ...interface{})
> Logf formats its arguments according to the format, analogous to Printf, and records the text in the error log. A final newline is added if not provided. For tests, the text will be printed only if the test fails or the -test.v flag is set. For benchmarks, the text is always printed to avoid having performance depend on the value of the -test.v flag.

But as we discussed earlier today, there is something blocking logs from streaming live to the stdout.